### PR TITLE
Error boundaries

### LINKED ISSUE #415: Set error boundaries for SmartFAQ
- Wrap the main App component in error boundary so we don't crash the whole frontend in case of render failure.
- Show error message if something went wrong.

More info: https://reactjs.org/docs/error-boundaries.html
Zeplin: https://zpl.io/aNpXrAZ

### PR DESCRIPTION
**Pull request by dinodsaurus**
closes #415 
~~~Also I need the proper fink for `Go to contact page`~~~
<br/><br/><br/><url>LiveURL: https://smartfaq-error-boundaries.surge.sh</url>

### CODE REVIEW

**Comment by jaroslav-kubicek** [comment]
> Also I need the proper fink for Go to contact page

it's same as for "Having trouble? Contact us"

https://www.kiwi.com/en/content/feedback

File: src/common/ErrorMessage.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
+// @flow
+/* eslint-disable react/no-unescaped-entities */
```
Comment (jaroslav-kubicek) [inline src/common/ErrorMessage.js]: why to disable this? :)

File: src/common/ErrorMessage.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      <Heading weight="bold">Something's not quite right.</Heading>
+      <div className="desc">
+        <Typography size="large" type="secondary">
+          Please, try refreshing the page. Sign out. Then sign in again. If the
+          problem persists, don't hesitate to contact us.
+        </Typography>
+      </div>
+      <p>
+        <TextLink
+          url="https://www.kiwi.com/en/content/feedback"
```
Comment (jaroslav-kubicek) [inline src/common/ErrorMessage.js]: see #428

File: src/common/ErrorMessage.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
+// @flow
```
Comment (jaroslav-kubicek) [inline src/common/ErrorMessage.js]: let's add snapshot test for this file, this component can't ever fail... :)

File: src/App.js [hunk 4]
```diff
@@ -88,31 +90,43 @@ class App extends React.PureComponent<Props, State> {
...
-                      setSelected={this.urlBookingSelected}
-                    />
-                    <Routes initialRoute={this.props.initialRoute} />
-                  </SearchStateProvider>
-                </BookingStateProvider>
-              </UserContext.Provider>
-            </CloseContext.Provider>
-          </LanguageContext.Provider>
-        </I18nextProvider>
+        <ErrorBoundaryProvider>
```
Comment (jaroslav-kubicek) [inline src/App.js]: why exactly is done using Context API, original example is much simpler:

https://reactjs.org/docs/error-boundaries.html#introducing-error-boundaries

```
if (this.state.hasError) {
  return <ErrorMessage />;
}

return  this.props.children;
```
and then:
```
<ErrorBoundary>
  <App {...props} />
</ErrorBoundary>
```

Comment (dinodsaurus) [reply src/App.js]: I thought to make it more "flexible" 😅 But I agree its a pre mature optimization...

File: src/common/ErrorMessage.js (hunk 1 above)
Comment (dinodsaurus) [reply src/common/ErrorMessage.js]: eslint is is complainging about `'`inside the text, do I need to enable it at again the end of the file?

**Pull request merged**